contracts/strategy/LiquidationStrategy.sol

```diff
@@ -230,180 +309,158 @@ contract LiquidationStrategy is BaseStrategy, ReentrancyGuard, IFlashLendingCall
+        } else {
+            uint256 _collateralAmountToLiquidateV2 = _vars.v2Ratio == 0 ? 0 : _collateralAmountToLiquidate.mul(_vars.v2Ratio).div(TENK);
+            uint256 balanceOfFXDBeforeSwap = fathomStablecoin.balanceOf(address(this));
+            if (_vars.v2Ratio > 0) {
+                fathomStablecoinReceivedV2 = _handleCollateralSellingV2(
+                    _vars,
+                    _collateralAmountToLiquidateV2,
+                    amountNeededToPayDebt.mul(_vars.v2Ratio).div(TENK)
+                );
+            }
+            if (_vars.v2Ratio < TENK) {
+                fathomStablecoinReceivedV3 = _sellCollateralV3(
+                    _vars.tokenAdapter.collateralToken(),
+                    address(fathomStablecoin),
+                    _collateralAmountToLiquidate.sub(_collateralAmountToLiquidateV2),
+                    uniswapV3Info[_vars.routerV3].poolFee,
+                    uniswapV3Info[_vars.routerV3].universalRouter
+                );
+            }
+            if ((fathomStablecoinReceivedV2 + fathomStablecoinReceivedV3) < amountNeededToPayDebt) {
+                if (fathomStablecoin.balanceOf(address(this)) < amountNeededToPayDebt) {
+                    revert NotEnoughToRepayDebt();
+                }
             }
-            // Deposit Fathom Stablecoin for liquidatorAddress
             _depositStablecoin(amountNeededToPayDebt, _vars.liquidatorAddress);
+            uint256 balanceOfFXDAfterSwap = fathomStablecoin.balanceOf(address(this));
             emit LogFlashLiquidationSuccess(
                 _vars.liquidatorAddress,
-                _debtValueToRepay,
+                amountNeededToPayDebt,
                 _collateralAmountToLiquidate,
-                dexAmountOut,
-                fathomStablecoinReceived,
-                path
+                fathomStablecoinReceivedV2,
+                fathomStablecoinReceivedV3,
+                _vars.v2Ratio
             );
+            if (balanceOfFXDBeforeSwap > balanceOfFXDAfterSwap) {
+                emit LogProfitOrLoss(balanceOfFXDBeforeSwap - balanceOfFXDAfterSwap, true);
+            } else {
+                emit LogProfitOrLoss(balanceOfFXDAfterSwap - balanceOfFXDBeforeSwap, false);
+            }
         }
     }
 
-    function _emergencyWithdraw(uint256 _amount) internal override {
-        require(_amount > 0, "LiquidationStrategy: zero amount");
-        require(_amount <= asset.balanceOf(address(this)), "LiquidationStrategy: wrong amount");
-        asset.safeTransfer(strategyManager, _amount);
-    }
-
     function supportsInterface(bytes4 _interfaceId) external pure returns (bool) {
         return type(IFlashLendingCallee).interfaceId == _interfaceId;
     }
 
+    function _nullifyIdleCollateral(address _collateral) internal {
+        if (idleCollateral[_collateral].CollateralAmount == 0) {
```

Comment:
since it's an internal fn we don't really need to have a check inside cause it will cause redundant sload call
it's better to check in functions that call `_nullifyIdleCollateral` to save gas, and maybe this functions will become redundant and will be replaced by `delete idleCollateral[_collateral]`